x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/advanced_step/advanced_step_details.tsx

```diff
@@ -22,7 +22,7 @@ import { ANALYSIS_CONFIG_TYPE } from '../../../../common/analytics';
 import { ANALYTICS_STEPS } from '../../page';
 
 function getStringValue(value: number | undefined) {
-  return value !== undefined ? `${value}` : UNSET_CONFIG_ITEM;
+  return value !== undefined && typeof value === 'number' ? `${value}` : UNSET_CONFIG_ITEM;
```

Comment:
nit, adding a typeof check here means the `undefined` check is redundant

Reply:
Good point - updated in 7e11c9e92fc44363b2433154ab7562bb55822eba